src/main/kotlin/de/cyface/collector/auth/MockedHandlerBuilder.kt

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2023 Cyface GmbH
+ *
+ * This file is part of the Cyface Data Collector.
+ *
+ * The Cyface Data Collector is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The Cyface Data Collector is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with the Cyface Data Collector. If not, see <http://www.gnu.org/licenses/>.
+ */
+package de.cyface.collector.auth
+
+import de.cyface.collector.handler.HTTPStatus
+import io.vertx.core.Future
+import io.vertx.core.json.JsonObject
+import io.vertx.ext.auth.impl.UserImpl
+import io.vertx.ext.web.Route
+import io.vertx.ext.web.RoutingContext
+import io.vertx.ext.web.handler.OAuth2AuthHandler
+import java.util.UUID
+
+/**
+ * Mocked OAuth2 builder which creates an OAuth2 handler for testing.
+ *
+ * @author Armin Schnabel
+ * @version 1.0.0
+ * @since 7.0.0
+ */
+class MockedHandlerBuilder : OAuth2HandlerBuilder {
+
+    override fun create(): Future<OAuth2AuthHandler> {
+        val handler: OAuth2AuthHandler = object : OAuth2AuthHandler {
+            override fun handle(event: RoutingContext) {
+                // FIXME: FileUploadTest ...withWrongCredentials expects this handler to return 401
+                // Testing the Mocked Auth Handler does not make much sense, so just remove the two tests (and this)?
+                val invalidCredentials = event.request().headers().get("Authorization").equals("Bearer invalidToken")
+                if (invalidCredentials) {
+                    event.fail(HTTPStatus.UNAUTHORIZED)
+                }
+
+                val principal = JsonObject()
+                    .put("username", "test-user")
+                    .put("sub", UUID.randomUUID()) // user id
+                val user = UserImpl(principal, JsonObject())
+                event.setUser(user)
+
+                // From AuthenticationHandlerImpl.handle @ `authenticate(ctx, authN -> {..})`
+                // event.session()?.regenerateId() - this leads to SessionExpired exception, thus, commented out
+                // proceed with the router
```

Comment:
This documents where I got the implementation structure of this mocked `OAuth2AuthHandler` from and why the session id reset was commented out. I'm not really sure why the original implementation does this, but don't really have the time right now to dive into this. But as this is only a mock, I think it's okay to just comment this out.